src/Api/include/Api/LockFreeApiEventProducer.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_API_LOCK_FREE_API_EVENT_PRODUCER_H_
+#define ORBIT_API_LOCK_FREE_API_EVENT_PRODUCER_H_
+
+#include "Api/EncodedEvent.h"
+#include "OrbitProducer/LockFreeBufferCaptureEventProducer.h"
+#include "ProducerSideChannel/ProducerSideChannel.h"
+
+namespace orbit_api {
+
+// This class is used to enqueue orbit_api::ApiEvent events from multiple threads and relay them to
+// OrbitService in the form of orbit_grpc_protos::ProducerCaptureEventCaptureEvent events.
+class LockFreeApiEventProducer
+    : public orbit_producer::LockFreeBufferCaptureEventProducer<orbit_api::ApiEvent> {
+ public:
+  LockFreeApiEventProducer() {
+    BuildAndStart(orbit_producer_side_channel::CreateProducerSideChannel());
+  }
+
+  ~LockFreeApiEventProducer() { ShutdownAndWait(); }
+
+ protected:
+  [[nodiscard]] virtual orbit_grpc_protos::ProducerCaptureEvent* TranslateIntermediateEvent(
+      orbit_api::ApiEvent&& raw_api_event, google::protobuf::Arena* arena) override {
```

Comment:
Maybe rename `raw_api_event` to just `api_event`.

Reply:
I want to make it clear that the intermediate events are "raw" events and not protos. Also api_event is used for another variable below.

Reply:
You're right, I had gotten this wrong.